pageserver/src/tenant.rs

```diff
@@ -849,7 +851,7 @@ impl Tenant {
                 // It also prevents the warmup proccess competing with the concurrency limit on
                 // logical size calculations: if logical size calculation semaphore is saturated,
                 // then warmup will wait for that before proceeding to the next tenant.
-                if let AttachType::Warmup(_permit) = attach_type {
+                if matches!(attach_type, AttachType::Warmup { during_startup: true, .. }) {
```

Comment:
:thinking: This change is not wrong, but interesting to think about what behaviour we want here.  When we do a storm of migrations to a pageserver, we can either let activation and logical size calculations proceed independently (current state in this PR), or force activations of Lazy tenants to wait for logical size calculation (i.e. change this condition to just `matches!(attach_type ,AttachType::Warmup{...}))`).

Probably makes sense to leave this as it is, and then when we're doing some HA testing (I'll be doing some of this for the storage controller soon), it'll be interesting to see how it behaves.